consensus/src/experimental/buffer.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::experimental::hashable::Hashable;
+use diem_crypto::HashValue;
+use std::collections::HashMap;
+
+pub struct LinkedItem<T: Hashable> {
+    // use option so we don't need T to be cloneable
+    elem: Option<T>,
+    // index is for find_element_by_key to have a starting position (similar to find_element)
+    index: u64,
+    next: Option<HashValue>,
+}
+
+pub type Link = Option<HashValue>;
```

Comment:
I find the name `Link` is confusing, Cursor or Handle seems better, what do you think?

Reply:
Agree. Let's change it to `Cursor`